src/form/acr.jl

```diff
@@ -47,6 +121,31 @@ function constraint_mc_voltage_magnitude_bounds(pm::_PM.AbstractACRModel, n::Int
     end
 end
 
+"bus voltage on/off constraint for load shed problem"
+function constraint_mc_bus_voltage_on_off(pm::_PM.AbstractACRModel; nw::Int=pm.cnw, kwargs...)
+    for (i,bus) in ref(pm, nw, :bus)
+        constraint_mc_bus_voltage_magnitude_on_off(pm, i; nw=nw)
+    end
+end
+
+
+"on/off bus voltage magnitude constraint"
+function constraint_mc_bus_voltage_magnitude_on_off(pm::_PM.AbstractACRModel, n::Int, i::Int, vmin, vmax)
+    vr = var(pm, n, :vr, i)
+    vi = var(pm, n, :vi, i)
+    z_voltage = var(pm, n, :z_voltage, i)
+
+    for c in conductor_ids(pm, n)
+        if isfinite(vmax[c])
+            JuMP.@constraint(pm.model, vr[c]^2 + vi[c]^2 <= vmax[c]^2*z_voltage)
+        end
+
+        if isfinite(vmin[c])
+            JuMP.@constraint(pm.model, vr[c]^2 + vi[c]^2 >= vmin[c]^2*z_voltage)
```

Comment:
Note, this is a non-convex constraint.  Take a note that we should look into ways to avoid it in the future.